File: angular/src/app/services/comment.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {CommentModel} from "../components/model/comment.model";
import {Observable} from "rxjs";
import {Comment} from "@angular/compiler";

@Injectable({
  providedIn: 'root'
})
export class CommentService {
  private baseUrl = "http://localhost:8585";

  constructor(
    private httpClient: HttpClient
  ) { }

  getComments(): Observable<CommentModel[]> {
    return this.httpClient.get<CommentModel[]>(`${this.baseUrl}/comments`);
  }

  saveComment(comment: CommentModel): Observable<CommentModel> {
    return this.httpClient.post<CommentModel>(`${this.baseUrl}/addcomment`, comment);
  }

  updateComment(comment: CommentModel): Observable<CommentModel> {
    return this.httpClient.put<CommentModel>(`${this.baseUrl}/comments/${comment.commentId}`, comment);
  }

  deleteComment(commentId: number): Observable<Object> {
    return this.httpClient.delete(`${this.baseUrl}/comments/${commentId}`)
  }
}

File: backend/src/main/java/com/saidj/trainingcenter/service/CommentService.java
package com.saidj.trainingcenter.service;

import java.util.List;

import com.saidj.trainingcenter.model.Comment;
import com.saidj.trainingcenter.model.Training;

public interface CommentService {
	List<Comment> getValidatedComments();
	Comment getComment(Long commentId);
	Comment saveComment(Comment comment);
	Comment updateComment(Comment comment);
	void deleteComment(Long commentId);
	
	Training getCommentTraining(Long commentId);
}
